### PR TITLE
WCS loading/saving enabled for dendrograms

### PR DESCRIPTION
**Pull request by tomr-stargazer**
Hi all (especially @ChrisBeaumont),

I decided I needed to see if dendrogram saving/loading with WCS objects (enabled at compute-time since #126) could be easily implemented. This is my quick first attempt at getting it to work. From my point of view it doesn't break too bad, but I haven't tested many cases, and admittedly I didn't try very hard to learn about the internals of the hdf5 module.

Especially let me know if I'm addressing this at the wrong level of abstraction.

### CODE REVIEW

**Comment by tomr-stargazer** [comment]
After playing around just a bit more, this code seems to be doing what I want - so I'll be using this in production. If desired I could try to add tests for wcs and non-wcs'd reloading.

**Comment by tomr-stargazer** [comment]
Also, there are weird Travis issues coming up:
https://travis-ci.org/dendrograms/astrodendro/jobs/54980582

```
    ----------------------------------------
    Command "/home/travis/miniconda/envs/test/bin/python -c "import setuptools, tokenize; __file__='/home/travis/build/dendrograms/astrodendro/src/astropy/setup.py'; exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" develop --no-deps" failed with error code 1 in /home/travis/build/dendrograms/astrodendro/src/astropy
The command "if [[ $ASTRO_VER == dev ]]; then pip install -e git+git://github.com/astropy/astropy.git#egg=astropy; else $CONDA_INSTALL astropy=$ASTRO_VER numpy=$NP_VER; fi" failed and exited with 1 during .
Your build has been stopped.
```

which seem version-dependent.

**Comment by ChrisBeaumont** [comment]
This looks good on the hdf5 side! It would be great to add a test. It also looks like the fits exporter/importer needs this patch.

**Comment by ChrisBeaumont** [comment]
@astrofrog what do you make about the jinja2 import error when pip installing astropy? https://travis-ci.org/dendrograms/astrodendro/jobs/54980585

We can probably fix this in astrodendro by installing jinja2 explicitly

**Comment by astrofrog** [comment]
Ah yes, we need to now include ``jinja2`` where we include Cython.

@tomr-stargazer - can you do this in this PR?

**Comment by astrofrog** [comment]
Basically we just need to add ``jinja2`` to ``CONDA_BASE``

**Comment by astrofrog** [comment]
Just for some background, this is due to the fact that astropy-dev now requires Cython *and* jinja2 to build.

**Comment by tomr-stargazer** [comment]
Hi @astrofrog  - I've just added jinja2 in the requisite place. (I haven't added @ChrisBeaumont's tests yet or any other changes so don't merge quite yet)

**Comment by tomr-stargazer** [comment]
Hi @ChrisBeaumont @astrofrog et al: 
I've just added the FITS functionality and tests for both hdf5 and fits saving/loading.
I think the tests should pass.
Do let me know if this is a clumsy implementation of saving a WCS header for the fits patch, as I'm not a fits/HDU pro.

File: astrodendro/io/hdf5.py [hunk 1]
```diff
@@ -41,21 +41,31 @@ def dendro_export_hdf5(d, filename):
...
+    except AttributeError:
+        pass
+
     f.close()
 
 
 def dendro_import_hdf5(filename):
     """Import 'filename' and construct a dendrogram from it"""
     import h5py
+    import astropy.wcs
```
Comment (ChrisBeaumont) [inline astrodendro/io/hdf5.py]: I have a slight preference for `from astropy.wcs import WCS` (a few less "wcs" to type)

File: astrodendro/io/fits.py [hunk 2]
```diff
@@ -42,13 +48,18 @@ def dendro_export_fits(d, filename):
 def dendro_import_fits(filename):
     """Import 'filename' and construct a dendrogram from it"""
     from astropy.io import fits
+    import astropy.wcs
```
Comment (ChrisBeaumont) [inline astrodendro/io/fits.py]: Slight preference for `from astropy.wcs import WCS`

File: astrodendro/tests/test_io.py [hunk 3]
```diff
@@ -111,6 +113,29 @@ def test_fits_auto(self):
         # recognize from signature
         d2 = Dendrogram.load_from('astrodendro-test')
 
+    def test_hdf5_with_wcs(self):
+        self.test_filename = 'astrodendro-test-wcs.hdf5'
+        test_wcs = WCS(header=dict(cdelt1=1, crval1=0, crpix1=1,
+                                 cdelt2=2, crval2=0, crpix2=1,
```
Comment (ChrisBeaumont) [inline astrodendro/tests/test_io.py]: It looks like the indentation here might not be pep-8 compliant

File: astrodendro/tests/test_io.py [hunk 4]
```diff
@@ -111,6 +113,29 @@ def test_fits_auto(self):
...
+        d1 = Dendrogram.compute(self.data, verbose=False, wcs=test_wcs)
+        d1.save_to(self.test_filename, format='hdf5')
+        d2 = Dendrogram.load_from(self.test_filename, format='hdf5')
+
+        assert d2.wcs.to_header_string() == d1.wcs.to_header_string()
+
+    def test_fits_with_wcs(self):
+        self.test_filename = 'astrodendro-test-wcs.fits'
+        test_wcs = WCS(header=dict(cdelt1=1, crval1=0, crpix1=1,
+                                 cdelt2=2, crval2=0, crpix2=1,
```
Comment (ChrisBeaumont) [inline astrodendro/tests/test_io.py]: same here

**Comment by ChrisBeaumont** [comment]
Looks good! There's a binary file in this PR `astrodendro/tests/astrodendro-test`, I assume that isn't intended?

File: astrodendro/tests/test_io.py (hunk 3 above)
Comment (tomr-stargazer) [inline astrodendro/tests/test_io.py]: oops! fixing it...
(@ChrisBeaumont Do you know how to get SublimeText to properly indent such sitations in python, automatically? I'm aware of plugins that will auto-format a whole file to pep8, but I find those too aggressive - usually I just want to be able to hit "tab" and have Sublime do the right thing. Emacs is better at this)

**Comment by tomr-stargazer** [comment]
@ChrisBeaumont Thanks for the comments, I've addressed each issue!

**Comment by tomr-stargazer** [comment]
@ChrisBeaumont @astrofrog 
Tests pass and I've addressed all issues, so I think this should be ready to merge, unless there are other concerns.

**Comment by ChrisBeaumont** [comment]
Looks good, thanks!

**Pull request merged**